File: JavaRecyclerViewApp/app/src/main/java/com/example/javarecyclerviewapp/MainActivity.java
package com.example.javarecyclerviewapp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.graphics.Color;
import android.os.Bundle;
import android.util.Log;
import android.widget.ImageView;

import com.squareup.picasso.Picasso;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private ArrayList<String> imageList = new ArrayList<>();
    private ArrayList<String> textList = new ArrayList<>();
    private ArrayList<Integer> colorList = new ArrayList<>();

    private ArrayList<ListItems> listItems = new ArrayList<>();

    private int listSize = 30;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initLists();
        initRecyclerView();
    }

    private void initLists()
    {

        for (int n = 1; n <= listSize; n++)
        {
            String url = "https://dummyimage.com/30x30/ffffff/000000.png&text=" + n;
            String txt = "Voici l'élément numéro " + n + " de cette liste";
            int color;
            if(n %3 == 1) {
                color = Color.BLUE;
            }
            else if (n % 3 == 2) {
                color = Color.GREEN;
            }
            else {
                color = Color.YELLOW;
            }
            listItems.add(new ListItems(url, txt, color));
        }
    }

    private void initRecyclerView(){
        RecyclerView recyclerView = findViewById(R.id.recyclerView);
        RecyclerViewAdapter adapter = new RecyclerViewAdapter(this, listItems);
        recyclerView.setAdapter(adapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
    }
}

File: JavaRecyclerViewApp/app/src/main/java/com/example/javarecyclerviewapp/MainActivity.kt
package com.example.javarecyclerviewapp

import android.graphics.Color
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import java.util.*

class MainActivity : AppCompatActivity() {

    private val listItems = ArrayList<ListItems>()

    private val listSize = 30

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initLists()
        initRecyclerView()
    }

    private fun initLists() {

        for (n in 1..listSize) {
            val url = "https://dummyimage.com/30x30/ffffff/000000.png&text=$n"
            val txt = "Voici l'élément numéro $n de cette liste"
            val color: Int = when {
                n % 3 == 1 -> Color.BLUE
                n % 3 == 2 -> Color.GREEN
                else -> Color.YELLOW
            }
            listItems.add(ListItems(url, txt, color))
        }
    }

    private fun initRecyclerView() {
        val recyclerView = findViewById<RecyclerView>(R.id.recyclerView)
        val adapter = RecyclerViewAdapter(this, listItems)
        recyclerView.adapter = adapter
        recyclerView.layoutManager = LinearLayoutManager(this)
    }
}